ayaneo-platform.c

```diff
@@ -394,29 +379,57 @@ static void ayaneo_led_mc_intensity(u8 group, u8 *color, u8 zones[])
 
 static void ayaneo_led_mc_off(void)
 {
-        ec_write_ram(AYANEO_LED_MC_ENABLE_ADDR, AYANEO_LED_MC_ENABLE_OFF);
-        ec_write_ram(AYANEO_LED_MC_ADDR_CLOSE_1, 0x01);
-        mdelay(AYANEO_LED_WRITE_DELAY_MS);
+        ayaneo_led_mc_set(AYANEO_LED_GROUP_LEFT,
+                AYANEO_LED_CMD_ENABLE_ADDR, AYANEO_LED_CMD_ENABLE_OFF);
+        ayaneo_led_mc_set(AYANEO_LED_GROUP_RIGHT,
+                AYANEO_LED_CMD_ENABLE_ADDR, AYANEO_LED_CMD_ENABLE_OFF);
 
         // note: omit for aya flip when implemented, causes unexpected behavior
         ayaneo_led_mc_set(AYANEO_LED_GROUP_LEFT_RIGHT, 0x00, 0x00);
 }
 
 static void ayaneo_led_mc_on(void)
 {
-        ec_write_ram(AYANEO_LED_MC_ENABLE_ADDR, AYANEO_LED_MC_ENABLE_ON);
-        ec_write_ram(AYANEO_LED_MC_ADDR_CLOSE_1, 0x01);
-        mdelay(AYANEO_LED_WRITE_DELAY_MS);
+        ayaneo_led_mc_set(AYANEO_LED_GROUP_LEFT,
+                AYANEO_LED_CMD_ENABLE_ADDR, AYANEO_LED_CMD_ENABLE_ON);
+        ayaneo_led_mc_set(AYANEO_LED_GROUP_RIGHT,
+                AYANEO_LED_CMD_ENABLE_ADDR, AYANEO_LED_CMD_ENABLE_ON);
+
+        ayaneo_led_mc_set(AYANEO_LED_GROUP_LEFT,
+                AYANEO_LED_CMD_PATTERN_ADDR, AYANEO_LED_CMD_PATTERN_OFF);
+        ayaneo_led_mc_set(AYANEO_LED_GROUP_RIGHT,
+                AYANEO_LED_CMD_PATTERN_ADDR, AYANEO_LED_CMD_PATTERN_OFF);
+
+        ayaneo_led_mc_set(AYANEO_LED_GROUP_LEFT,
+                AYANEO_LED_CMD_FADE_ADDR, AYANEO_LED_CMD_FADE_OFF);
+        ayaneo_led_mc_set(AYANEO_LED_GROUP_RIGHT,
+                AYANEO_LED_CMD_FADE_ADDR, AYANEO_LED_CMD_FADE_OFF);
+
+        // set each sector to RGB mode
+        ayaneo_led_mc_set(AYANEO_LED_GROUP_LEFT, 0x11, 0x05);
```

Comment:
It's still a bit ambiguous unfortunately; not sure what to call it...

From the usages it looks like in the context of the 0x11-0x14 commands the second byte is bitflags indicating on/off where the top 4 bits are the sectors of the left stick and the bottom 4 the right one, but I haven't been able to fully decipher it without the hardware in hand to try the different values and observe the results.
There seems to be 2 different meanings but I haven't been able to figure out how it knows which way to interpret it.

For example, in the scan animation, think Kit from knight rider, the sequence is:
0x11, 0x81 (0b10000001)
0x12, 0x42 (0b01000010)
0x13, 0x24 (0b00100100)
0x14, 0x18 (0b00011000)

I kept the 0x05 value from the original string of bytes that was here for these devices since it aligns with what I've seen for the static on mode.

Reply:
Maybe AYANEO_LED_MODE_STATIC or something similar will work for that

Reply:
Introduced consts :)